cloudferrylib/os/image/glance_image.py

```diff
@@ -450,6 +450,21 @@ def deploy(self, info):
         LOG.info("Glance images deployment finished.")
         return new_info
 
+    def update_membership(self, image_id, image_members):
+        client = self.glance_client
+        existing_members = {m.member_id: m
+                            for m in client.image_members.list(image=image_id)}
+        for tenant_name, can_share in image_members.iteritems():
+            tenant_id = self.identity_client.get_tenant_id_by_name(tenant_name)
+            LOG.debug("Deploying image member for image '%s' "
+                      "tenant '%s'", image_id, tenant_name)
+            if (tenant_id in existing_members and
+                    existing_members[tenant_id].can_share != can_share):
+                client.image_members.delete(image_id, tenant_id)
+                del existing_members[tenant_id]
```

Comment:
`del` doesn't look good in most contexts (in this too). Perhaps `filter()` or `set()` of tenant IDs instead?